### PR TITLE
Add public ingress url to allowed return urls

### PR DESCRIPTION
**Pull request by bencekov**
When the public ingress is available, its url is added to the allowed return urls in Kratos' config file.

If the url of the public ingress is http://ingress:80/path/subpath it gets added to the config file as https://ingress/*.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -340,12 +340,19 @@ def _render_conf_file(self) -> str:
 
         default_schema_id, schemas = self._get_identity_schema_config()
         login_ui_url = self._get_login_ui_endpoint_info("login_url")
+
+        allowed_return_urls = []
+        if self._public_url:
+            allowed_return_urls.append(domain_url(self._public_url))
+        if login_ui_url:
+            allowed_return_urls.append(login_ui_url)
```
Comment (nsklikas) [inline src/charm.py:348]: I don't think this is needed anymore (the login UI is in the same domain)

Comment (bencekov) [reply src/charm.py:348]: true, thanks for the catch!

File: src/charm.py [hunk 2]
```diff
@@ -361,12 +361,18 @@ def _render_conf_file(self) -> str:
         login_ui_url = self._get_login_ui_endpoint_info("login_url")
         mappers = self._get_claims_mappers()
         cookie_secrets = self._get_secret()
+
+        allowed_return_urls = []
+        if self._public_url:
+            allowed_return_urls.append(domain_url(self._public_url))
```
Comment (shipperizer) [inline src/charm.py:367]:
```suggestion
            allowed_return_urls.append(urlparse(self.public_url)._replace(path="*", params="", query="", fragment="").geturl())
```


I would put it inline, jumping across module for a function that is a little wrapper of a built-in function doesn't seem that intuitive

File: src/charm.py [hunk 3]
```diff
@@ -361,12 +362,22 @@ def _render_conf_file(self) -> str:
...
+
         rendered = template.render(
             cookie_secrets=[cookie_secrets] if cookie_secrets else None,
             log_level=self._log_level,
             mappers=mappers,
-            default_browser_return_url=self._get_login_ui_endpoint_info("login_url"),
-            allowed_return_urls=[login_ui_url] if login_ui_url else [],
+            mappers_path=str(self._mappers_dir_path),
+            default_browser_return_url=login_ui_url,
+            allowed_return_urls=allowed_return_urls,
```
Comment (nsklikas) [inline src/charm.py:380]:
```suggestion
            default_browser_return_url=login_ui_url,
            allowed_return_urls=allowed_return_urls,
```

**Comment by shipperizer** [comment]
there are 4 commits and 16 additions, which are not really distinguishable/easy to separate

I'd say make one commit out of it

File: src/charm.py (hunk 3 above)
Comment (bencekov) [reply src/charm.py]: done

**Review by nsklikas — APPROVED** [review]

**Pull request merged**